caluma/core/models.py

```diff
@@ -1,9 +1,40 @@
 import uuid
 
+from django.conf import settings
 from django.db import models
+from simple_history.models import HistoricalRecords
 
 
-class BaseModel(models.Model):
+def _history_user_getter(historical_instance):
+    return historical_instance.history_user_id
+
+
+def _history_user_setter(historical_instance, user):
+    request = getattr(HistoricalRecords.thread, "request", None)
+    if request is not None:
+        historical_instance.history_user_id = request.user.username
+    else:
+        historical_instance.history_user_id = None
+
+
+class HistoryModel(models.Model):
+    history = HistoricalRecords(
+        inherit=True,
+        history_user_id_field=models.CharField(null=True, max_length=150),
+        history_user_setter=_history_user_setter,
+        history_user_getter=_history_user_getter,
+    )
+
+    class Meta:
+        abstract = True
+
+
+Base = models.Model
+if settings.SIMPLE_HISTORY_ACTIVE:
```

Comment:
What does this do with regards to the migrations? Will enabling this at any time create any tables, and does it do so correctly? Or will it end up requiring additional migrations?

Reply:
When switching `SIMPLE_HISTORY_ACTIVE`, it's necessary to run `makemigrations` and `migrate`.

Reply:
Unfortunately there seems to be no "master switch". There is [save-without-a-historical-record](https://django-simple-history.readthedocs.io/en/2.7.2/querying_history.html#save-without-a-historical-record), but that's not really what we need.

Reply:
> When switching `SIMPLE_HISTORY_ACTIVE`, it's necessary to run `makemigrations` and `migrate`.

That's kind of a killer. If a client decides to switch, they can't expect to go into the container and do that manually.

I'm OK with switching on/off not being possible, but we need to instruct users properly. Consider the following cases:

| Project status     | Want the feature?    |   Action required
|----------------|-------------------------|-----------------------------------|
| New           |  No                          |   None                                |
| New           |  Yes                          |   Set the flag correctly from start  |
| Existing      |  No                           | None                                 |
| Existing      |  Yes                          | Set the flag exactly at the same time when Caluma is updated |

All this trouble and possible errors by operators make me consider simply not providing the switch and enabling the feature permanently

Reply:
I absolutely agree. The switch is removed.